### ISSUE TITLE
Release builds should generate a PDB file to enable better crash reporting.

### ISSUE THREAD

**Issue by someone972** [Regalis11/Barotrauma#4105]
- [X] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Currently, the release version of Barotrauma does not include a PDB file. This causes the crash dumps to only include the function which failed, without a line number. By setting the debugging information option to 'pdb-only', and including the generated PDB in the release, it would greatly enhance the debugability of issues found on the release version. This is especially true since the codebase has a lot of lengthy functions where just knowing the function name often isn't enough. The PDB file only includes source file/line number/ local variable names, which shouldn't be an issue given the mostly open-source nature of the application. 

Enabling pdb-only for release would:
- Still apply all the optimizations that the release build would usually get
- Won't affect performance of the running application (should only access PDB when accessing StackTrace)
- Enable line numbers in crash dumps

I also did some tests to verify that the information I based this suggestion on was correct. I compared the build size before and after enabling pdb output, as well as the decompiled output with dotPeek. With optimizations on, the build size barely changed and the decompiled output showed they were nearly identical with extremely minor differences. With optimizations off (note that DEBUG is still not defined) the size grew significantly with many differences.
Some statistics:
- Debug mode: none | Debug symbols: false | Optimizations: true -> Barotrauma.dll 4726 KB
- Debug mode: full | Debug symbols: true | Optimizations: true -> Barotrauma.dll 4746 KB
- Debug mode: full | Debug symbols: true | Optimizations: false -> Barotrauma.dll 5131 KB

In summary, I believe enabling and distributing the PDB file on release builds would greatly improve the ability to triage issues without having any significant downsides. If there is some other business case which I had not considered, please let me know.

**Comment by Regalis11** [comment]
The game does generate PDB files in release builds, looks like we accidentally left them out while uploading the latest builds to Steam (SteamPipe ignores them by default, and it seems we accidentally went with the default settings in the past few builds).

**Comment by someone972** [comment]
Well looks like I didn't need to do all that work to try and convince you then 😄 . Oh well, I increased my own understanding of how PDB files work so it's all good.
